coffee-chats/src/main/java/com/google/step/coffee/entity/ChatRequestBuilder.java

```diff
@@ -0,0 +1,67 @@
+package com.google.step.coffee.entity;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class ChatRequestBuilder {
+  private static final int MAX_PARTICIPANTS = 5;
+
+  private List<String> tags = new ArrayList<>();
+  private List<Date> dates = new ArrayList<>();
+  private int minPeople = 1;
+  private int maxPeople = 4;
+  private int duration = 30;
+  private boolean matchRandom = false;
+  private boolean matchRecents = true;
+
+  public ChatRequest build() {
+    if (dates.isEmpty()) {
+      throw new IllegalStateException("At least one date must be set");
+    }
+    return new ChatRequest(tags, dates, minPeople, maxPeople, duration, matchRandom, matchRecents);
+  }
+
+  public ChatRequestBuilder withTags(List<String> tags) {
+    this.tags = tags;
+    return this;
+  }
+
+  public ChatRequestBuilder onDates(List<Date> dates) {
+    if (!dates.isEmpty()) {
+      this.dates = dates;
+      return this;
+    } else {
+      throw new IllegalArgumentException("No dates selected");
+    }
+  }
+
+  public ChatRequestBuilder withGroupSize(int minPeople, int maxPeople) {
+    if (minPeople > 0 && minPeople <= maxPeople && maxPeople < MAX_PARTICIPANTS) {
```

Comment:
Can minPeople be 1? How would that work?

Is maxPeople < (not <=) max_participating intended?

Reply:
I should write some comments here clarifying the intended use of the parameter. This is the number of other users this request is attempting to match to i.e. 1 implies 1 other person so a meeting size of 2, 4 implies a meeting size of 5. I shall clarify this with some comments.